src/note.rs

```diff
@@ -113,10 +113,14 @@ impl Note {
         nonce: JubJubScalar,
         psk: &PublicSpendKey,
         value: u64,
-        blinding_factor: JubJubScalar,
+        mut blinding_factor: JubJubScalar,
     ) -> Self {
         let stealth_address = psk.gen_stealth_address(r);
 
+        if let NoteType::Transparent = note_type {
+            blinding_factor = JubJubScalar::zero();
+        }
```

Comment:
This should be therefore removed (as mentioned, a debug assertion might be useful instead).

The part where you want to put this code is instead in `new`:
```rust
    pub fn new<R: RngCore + CryptoRng>(
        rng: &mut R,
        note_type: NoteType,
        psk: &PublicSpendKey,
        value: u64,
    ) -> Self {
        let r = JubJubScalar::random(rng);
        let nonce = JubJubScalar::random(rng);
        let blinding_factor = if let NoteType::Transparent = note_type {
            JubJubScalar::zero()
        } else {
            JubJubScalar::random(rng)
        };

        Self::deterministic(note_type, &r, nonce, psk, value, blinding_factor)
    }
```

Reply:
Fixed